pkg/logs/assetvalidator_error.go

```diff
@@ -20,10 +22,10 @@ type AssetValidationError struct {
 
 // ValidationErrorLogFactory is a factory function that takes an error that occurs during validation,
 // and returns a corresponding struct with logging information
-func ValidationErrorLogFactory(validationErr error, assetID int64) interface{} {
+func ValidationErrorLogFactory(validationErr error, assetID int64, arn string, resourceType string) interface{} {
 	switch validationErr.(type) {
 	case domain.ValidationFailure:
-		return AssetValidationFailure{Reason: validationErr.Error(), AssetID: assetID}
+		return AssetValidationFailure{Reason: validationErr.Error(), AssetID: assetID, ARN: arn, ResourceType: resourceType}
```

Comment:
Stream of consciousness: In such a sense, `ARN` and `ResourceType` shouldn't be in OSS. Amazon-specific properties shouldn't be leaked in a "generic attributor." In such a sense, asset-attributor should not be concerned with where it's pulling information, whether it's Amazon specific, GCP, Azure, etc. Thus, I'm tempted to say let's move this factory function to closed source. However, the Attribute Handler is open sourced. With that said, we could just not log `ARN` and `ResourceType`, and bundle it in our `Reason` field, but that could clutter our logs. For now, i'm iffy on this one, unless I can be convinced otherwise.

Reply:
In such a sense, I kinda disagree with the notion of ARN being in `CloudAssetDetails` in the first place, as shouldn't be amazon specific anyways, no? I don't think ARN pertains to IPAM assets too...

Reply:
Okay cool, I see your point. I'll try to make these changes in the close source instead.